app/services/support_interface/unexplained_breaks_in_work_history_export.rb

```diff
@@ -0,0 +1,108 @@
+module SupportInterface
+  class UnexplainedBreaksInWorkHistoryExport
+    class UnexplainedBreak
+      def initialize(month_range:)
+        @month_range = month_range
+      end
+
+      def start_date
+        @month_range.first.prev_month
+      end
+
+      def end_date
+        @month_range.last.next_month
+      end
+
+      def length
+        ((end_date.year * 12) + end_date.month) - ((start_date.year * 12) + start_date.month)
+      end
+    end
+
+    def data_for_export
+      data_for_export = Candidate.all.map do |candidate|
+        application_form = candidate.application_forms.order(:submitted_at).last
+        next if application_form.nil?
+
+        unexplained_breaks = get_unexplained_breaks(application_form)
+        next if unexplained_breaks.nil?
+
+        total_unexplained_time = unexplained_breaks.sum(&:length)
+        output = {
+            'Candidate id' => application_form.candidate_id,
+            'Application id' => application_form.id,
+            'Start of working life' => get_start_of_working_life(application_form),
+            'Number of unexplained breaks' => unexplained_breaks.length,
+            'Total unexplained time (months)' => total_unexplained_time
+        }
+        output
+      end
+
+      data_for_export.compact
+    end
+
+    private
+
+    def get_unexplained_breaks(application_form)
+      work_history = application_form.application_work_experiences.sort_by(&:start_date)
+      existing_breaks = application_form.application_work_history_breaks.sort_by(&:start_date)
+
+      work_history_with_breaks = []
+      work_history.each { |job| work_history_with_breaks << job }
+      existing_breaks.each { |existing_break| work_history_with_breaks << existing_break }
+
+      if work_history.any?
+        timeline_in_months = month_range(
+            start_date: get_start_of_working_life(application_form),
+            end_date: submitted_at(application_form) - 1.month,
+        )
+        break_months_in_timeline = remove_months(timeline: timeline_in_months, entries: work_history, application_form: application_form)
+        remaining_months = remove_months(timeline: break_months_in_timeline, entries: existing_breaks, application_form: application_form)
+
+        create_unexplained_breaks(remaining_months)
+      end
+    end
+
+    def get_start_of_working_life(application_form)
+      application_form.date_of_birth.beginning_of_month + 18.years
```

Comment:
The `.beginning_of_month` was to sanitise the data so that birthdays don't leak in to the export, since we don't need it for the calculations.

I'd be interested to know if you think this step was necessary.

Reply:
Makes sense to me but I'd probably run that one passed Mike Emery as it sounds more like a policy / decision rule

Reply:
I spoke to Mike and he thought it was a good idea 👍